ml-agents/mlagents/trainers/policy/policy.py

```diff
@@ -49,12 +49,7 @@ def __init__(
             1 for shape in behavior_spec.observation_shapes if len(shape) == 3
         )
         self.use_continuous_act = self.behavior_spec.action_spec.is_continuous()
-        # This line will be removed in the ActionBuffer change
-        self.num_branches = (
-            self.behavior_spec.action_spec.continuous_size
-            + self.behavior_spec.action_spec.discrete_size
-        )
-        self.previous_action_dict: Dict[str, np.array] = {}
+        self.previous_action_dict: Dict[str, Dict[str, np.ndarray]] = {}
```

Comment:
Why not `Dict[str, ActionTuple]`? a dictionary with two hardcoded keys (in `make_empty_previous_action()`) doesn't make much sense.

Reply:
Good call out. I was doing this because it made writing to the buffer easier (e.g. using the string) but in reality we only ever use the previous action if its discrete. I've updated the PR to only save the discrete actions.